src/flask_migrate/templates/flask/env.py

```diff
@@ -29,7 +29,9 @@ def get_engine():
 # from myapp import mymodel
 # target_metadata = mymodel.Base.metadata
 config.set_main_option(
-    'sqlalchemy.url', str(get_engine().url).replace('%', '%%'))
+    "sqlalchemy.url",
+    str(get_engine().url.render_as_string(hide_password=False)).replace("%", "%%"),
```

Comment:
Is the `str()` necessary here?
Does this work with SQLAlchemy 1.4, 1.3, and older? Or is it only 2.0, which would make this change break with older versions?

Reply:
Yes the version 1.4 it's okey but 1.3 I don't think so. We can put it in **get_engine** function and catch the error if the attribute of render_as_string does not exist for the older version users. What do you think ? Because I don't know there is a way to retrieve the current version of SQLAlchemy...

Reply:
By the way, str() already existed so I didn't remove it

Reply:
I know, but `str()` was necessary before, and now seems that it isn't doing anything right? The result from `render_as_string()` is already a string, correct?

Reply:
sqlalchemy is not the best at naming thing, but at least it manages to return a string from a `render_as_string` method :)